nixos/modules/services/hardware/udev.nix

```diff
@@ -202,12 +202,22 @@ in
         '';
       };
 
-      extraRules = mkOption {
+      initrdRules = mkOption {
         default = "";
         example = ''
           SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="00:1D:60:B9:6D:4F", KERNEL=="eth*", NAME="my_fast_network_card"
         '';
         type = types.lines;
+        description = ''
+          <command>udev</command> rules to include in the initrd. They'll
+          be written into file <filename>99-local.rules</filename>. Thus
+          they are read and applied after all other rules.
```

Comment:
Are they also added to the udev rules on the real system?

What does "all other rules" here refer to?

Reply:
There's a bunch of hardcoded rules in nixos/modules/system/boot/stage-1.nix. Mostly udev, lvm and some display manager thing.

Reply:
Okay, so the initrd udev rules are *only* added to the initrd, not the main system?

Reply:
Correct, I'll write it more clearly.